\documentclass[]{article}
\usepackage{lmodern}
\usepackage{amssymb,amsmath}
\usepackage{ifxetex,ifluatex}
\usepackage{fixltx2e} % provides \textsubscript
\ifnum 0\ifxetex 1\fi\ifluatex 1\fi=0 % if pdftex
  \usepackage[T1]{fontenc}
  \usepackage[utf8]{inputenc}
\else % if luatex or xelatex
  \ifxetex
    \usepackage{mathspec}
  \else
    \usepackage{fontspec}
  \fi
  \defaultfontfeatures{Ligatures=TeX,Scale=MatchLowercase}
\fi
% use upquote if available, for straight quotes in verbatim environments
\IfFileExists{upquote.sty}{\usepackage{upquote}}{}
% use microtype if available
\IfFileExists{microtype.sty}{%
\usepackage{microtype}
\UseMicrotypeSet[protrusion]{basicmath} % disable protrusion for tt fonts
}{}


\usepackage{longtable,booktabs}
\usepackage{graphicx}
% grffile has become a legacy package: https://ctan.org/pkg/grffile
\IfFileExists{grffile.sty}{%
\usepackage{grffile}
}{}
\makeatletter
\def\maxwidth{\ifdim\Gin@nat@width>\linewidth\linewidth\else\Gin@nat@width\fi}
\def\maxheight{\ifdim\Gin@nat@height>\textheight\textheight\else\Gin@nat@height\fi}
\makeatother
% Scale images if necessary, so that they will not overflow the page
% margins by default, and it is still possible to overwrite the defaults
% using explicit options in \includegraphics[width, height, ...]{}
\setkeys{Gin}{width=\maxwidth,height=\maxheight,keepaspectratio}
\IfFileExists{parskip.sty}{%
\usepackage{parskip}
}{% else
\setlength{\parindent}{0pt}
\setlength{\parskip}{6pt plus 2pt minus 1pt}
}
\setlength{\emergencystretch}{3em}  % prevent overfull lines
\providecommand{\tightlist}{%
  \setlength{\itemsep}{0pt}\setlength{\parskip}{0pt}}
\setcounter{secnumdepth}{5}

%%% Use protect on footnotes to avoid problems with footnotes in titles
\let\rmarkdownfootnote\footnote%
\def\footnote{\protect\rmarkdownfootnote}

%%% Change title format to be more compact
\usepackage{titling}

% Create subtitle command for use in maketitle
\providecommand{\subtitle}[1]{
  \posttitle{
    \begin{center}\large#1\end{center}
    }
}

\setlength{\droptitle}{-2em}

\RequirePackage[]{C:/Users/Sarah/Documents/R/win-library/4.1/BiocStyle/resources/tex/Bioconductor}

\bioctitle[]{CTDPathSim2.0 Vignette}
    \pretitle{\vspace{\droptitle}\centering\huge}
  \posttitle{\par}
\author{By Banabithi Bose}
    \preauthor{\centering\large\emph}
  \postauthor{\par}
      \predate{\centering\large\emph}
  \postdate{\par}
    \date{4/10/2022}

% code highlighting
\definecolor{fgcolor}{rgb}{0.251, 0.251, 0.251}
\newcommand{\hlnum}[1]{\textcolor[rgb]{0.816,0.125,0.439}{#1}}%
\newcommand{\hlstr}[1]{\textcolor[rgb]{0.251,0.627,0.251}{#1}}%
\newcommand{\hlcom}[1]{\textcolor[rgb]{0.502,0.502,0.502}{\textit{#1}}}%
\newcommand{\hlopt}[1]{\textcolor[rgb]{0,0,0}{#1}}%
\newcommand{\hlstd}[1]{\textcolor[rgb]{0.251,0.251,0.251}{#1}}%
\newcommand{\hlkwa}[1]{\textcolor[rgb]{0.125,0.125,0.941}{#1}}%
\newcommand{\hlkwb}[1]{\textcolor[rgb]{0,0,0}{#1}}%
\newcommand{\hlkwc}[1]{\textcolor[rgb]{0.251,0.251,0.251}{#1}}%
\newcommand{\hlkwd}[1]{\textcolor[rgb]{0.878,0.439,0.125}{#1}}%
\let\hlipl\hlkwb
%
\usepackage{fancyvrb}
\newcommand{\VerbBar}{|}
\newcommand{\VERB}{\Verb[commandchars=\\\{\}]}
\DefineVerbatimEnvironment{Highlighting}{Verbatim}{commandchars=\\\{\}}
%
\newenvironment{Shaded}{\begin{myshaded}}{\end{myshaded}}
% set background for result chunks
\let\oldverbatim\verbatim
\renewenvironment{verbatim}{\color{codecolor}\begin{myshaded}\begin{oldverbatim}}{\end{oldverbatim}\end{myshaded}}
%
\newcommand{\KeywordTok}[1]{\hlkwd{#1}}
\newcommand{\DataTypeTok}[1]{\hlkwc{#1}}
\newcommand{\DecValTok}[1]{\hlnum{#1}}
\newcommand{\BaseNTok}[1]{\hlnum{#1}}
\newcommand{\FloatTok}[1]{\hlnum{#1}}
\newcommand{\ConstantTok}[1]{\hlnum{#1}}
\newcommand{\CharTok}[1]{\hlstr{#1}}
\newcommand{\SpecialCharTok}[1]{\hlstr{#1}}
\newcommand{\StringTok}[1]{\hlstr{#1}}
\newcommand{\VerbatimStringTok}[1]{\hlstr{#1}}
\newcommand{\SpecialStringTok}[1]{\hlstr{#1}}
\newcommand{\ImportTok}[1]{{#1}}
\newcommand{\CommentTok}[1]{\hlcom{#1}}
\newcommand{\DocumentationTok}[1]{\hlcom{#1}}
\newcommand{\AnnotationTok}[1]{\hlcom{#1}}
\newcommand{\CommentVarTok}[1]{\hlcom{#1}}
\newcommand{\OtherTok}[1]{{#1}}
\newcommand{\FunctionTok}[1]{\hlstd{#1}}
\newcommand{\VariableTok}[1]{\hlstd{#1}}
\newcommand{\ControlFlowTok}[1]{\hlkwd{#1}}
\newcommand{\OperatorTok}[1]{\hlopt{#1}}
\newcommand{\BuiltInTok}[1]{{#1}}
\newcommand{\ExtensionTok}[1]{{#1}}
\newcommand{\PreprocessorTok}[1]{\textit{#1}}
\newcommand{\AttributeTok}[1]{{#1}}
\newcommand{\RegionMarkerTok}[1]{{#1}}
\newcommand{\InformationTok}[1]{\textcolor{messagecolor}{#1}}
\newcommand{\WarningTok}[1]{\textcolor{warningcolor}{#1}}
\newcommand{\AlertTok}[1]{\textcolor{errorcolor}{#1}}
\newcommand{\ErrorTok}[1]{\textcolor{errorcolor}{#1}}
\newcommand{\NormalTok}[1]{\hlstd{#1}}
%
\AtBeginDocument{\bibliographystyle{C:/Users/Sarah/Documents/R/win-library/4.1/BiocStyle/resources/tex/unsrturl}}


\begin{document}
\maketitle


{
\setcounter{tocdepth}{2}
\tableofcontents
\newpage
}
\hypertarget{introduction}{%
\section{\texorpdfstring{ Introduction}{ Introduction}}\label{introduction}}

CTDPathSim2.0 is a computational tool which utilizes a pathway activity-based approach to compute similarity scores between primary tumor samples and cell lines. CTDPathSim2.0 integrates DNA methylation, gene expression and copy number variation datasets. The package is able to run without copy number variation to produce the results based on previous version of the tool, CTDPathSim1.0. CTDPathSim2.0 has five main computational steps:

Step 1:Computing sample-specific deconvoluted DNA methylation profile

Step 2:Computing sample-specific deconvoluted expression profile

Step 3:Computing sample and cell line-specific differentially expressed (DE) genes and enriched biological pathways

Step 4:Computing sample and cell line-specific differentially methylated (DM) and differentially aberrated (DA) genes

Step 5:Computing sample-cell line pathway activity-based similarity score

\begin{Shaded}
\begin{Highlighting}[]
\FunctionTok{library}\NormalTok{(CTDPathSim2}\FloatTok{.0}\NormalTok{)}
\FunctionTok{library}\NormalTok{(printr)}
\FunctionTok{data}\NormalTok{(}\StringTok{"CTDPathSim2"}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\hypertarget{example-data-sets}{%
\section{\texorpdfstring{ Example Data Sets}{  Example Data Sets}}\label{example-data-sets}}

\hypertarget{methylation.sample}{%
\subsection{\texorpdfstring{ Methylation.Sample}{  Methylation.Sample}}\label{methylation.sample}}

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{knitr}\SpecialCharTok{::}\FunctionTok{kable}\NormalTok{(Methylation.Sample[}\DecValTok{1}\SpecialCharTok{:}\DecValTok{5}\NormalTok{,}\DecValTok{1}\SpecialCharTok{:}\DecValTok{3}\NormalTok{], }\AttributeTok{digits =} \DecValTok{2}\NormalTok{, }\AttributeTok{caption =} \StringTok{\textquotesingle{}Probe{-}centric DNA methylation values of the tumor samples.\textquotesingle{}}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\begin{table}

\caption{\label{tab:unnamed-chunk-58}Probe-centric DNA methylation values of the tumor samples.}
\centering
\begin{tabular}[t]{l|r|r|r}
\hline
  & TCGA-OU-A5PI-01A & TCGA-OR-A5K8-01A & TCGA-OR-A5JX-01A\\
\hline
cg00323915 & 0.93 & 0.55 & 0.32\\
\hline
cg21830221 & 0.87 & 0.90 & 0.63\\
\hline
cg11348106 & 0.91 & 0.55 & 0.77\\
\hline
cg12232731 & 0.82 & 0.30 & 0.55\\
\hline
cg07218880 & 0.16 & 0.13 & 0.07\\
\hline
\end{tabular}
\end{table}

\hypertarget{reference.methylation.markers}{%
\subsection{\texorpdfstring{ Reference.Methylation.Markers}{  Reference.Methylation.Markers}}\label{reference.methylation.markers}}

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{knitr}\SpecialCharTok{::}\FunctionTok{kable}\NormalTok{(Reference.Methylation.Markers[}\DecValTok{1}\SpecialCharTok{:}\DecValTok{5}\NormalTok{], }\AttributeTok{caption =} \StringTok{\textquotesingle{}The marker loci (probes) of reference cell types.\textquotesingle{}}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\begin{table}

\caption{\label{tab:unnamed-chunk-59}The marker loci (probes) of reference cell types.}
\centering
\begin{tabular}[t]{l}
\hline
x\\
\hline
cg00323915\\
\hline
cg21830221\\
\hline
cg11348106\\
\hline
cg12232731\\
\hline
cg07218880\\
\hline
\end{tabular}
\end{table}

\hypertarget{reference.methylation.celltypes}{%
\subsection{\texorpdfstring{ Reference.Methylation.CellTypes}{  Reference.Methylation.CellTypes}}\label{reference.methylation.celltypes}}

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{knitr}\SpecialCharTok{::}\FunctionTok{kable}\NormalTok{(Reference.Methylation.CellTypes[}\DecValTok{1}\SpecialCharTok{:}\DecValTok{3}\NormalTok{,}\DecValTok{1}\SpecialCharTok{:}\DecValTok{3}\NormalTok{], }\AttributeTok{caption =} \StringTok{\textquotesingle{}R dataframe for the DNA methylation values for the reference cell types and the probes.\textquotesingle{}}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\begin{table}

\caption{\label{tab:unnamed-chunk-60}R dataframe for the DNA methylation values for the reference cell types and the probes.}
\centering
\begin{tabular}[t]{l|r|r|r}
\hline
  & Neu & NK & Neu\_1\\
\hline
cg00323915 & 0.8715876 & 0.2367720 & 0.8266145\\
\hline
cg21830221 & 0.8344109 & 0.0714415 & 0.8273859\\
\hline
cg11348106 & 0.8878073 & 0.4177928 & 0.8717599\\
\hline
\end{tabular}
\end{table}

\hypertarget{annotation27k}{%
\subsection{\texorpdfstring{ Annotation27K}{  Annotation27K}}\label{annotation27k}}

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{knitr}\SpecialCharTok{::}\FunctionTok{kable}\NormalTok{(Annotation27K[}\DecValTok{1}\SpecialCharTok{:}\DecValTok{3}\NormalTok{,], }\AttributeTok{caption =} \StringTok{\textquotesingle{}An R dataframe object with probe, symbol and synonym columns of 27K DNA methylation data annotation.\textquotesingle{}}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\begin{table}

\caption{\label{tab:unnamed-chunk-61}An R dataframe object with probe, symbol and synonym columns of 27K DNA methylation data annotation.}
\centering
\begin{tabular}[t]{l|l|l}
\hline
ID & Symbol & Synonym\\
\hline
cg00000292 & ATP2A1 & ATP2A; SERCA1;\\
\hline
cg00002426 & SLMAP & SLAP; KIAA1601;\\
\hline
cg00003994 & MEOX2 & GAX; MOX2;\\
\hline
\end{tabular}
\end{table}

\hypertarget{annotation450k}{%
\subsection{\texorpdfstring{ Annotation450K}{  Annotation450K}}\label{annotation450k}}

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{knitr}\SpecialCharTok{::}\FunctionTok{kable}\NormalTok{(Annotation450K[}\FunctionTok{c}\NormalTok{(}\DecValTok{6}\NormalTok{,}\DecValTok{7}\NormalTok{,}\DecValTok{10}\NormalTok{),], }\AttributeTok{caption =} \StringTok{\textquotesingle{}An R dataframe object with probe ID, UCSC\_REFGENE\_NAME and UCSC\_REFGENE\_GROUP columns 450K DNA methylation data annotation.\textquotesingle{}}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\textbackslash begin\{table\}

\textbackslash caption\{\label{tab:unnamed-chunk-62}An R dataframe object with probe ID, UCSC\_REFGENE\_NAME and UCSC\_REFGENE\_GROUP columns 450K DNA methylation data annotation.\}
\centering

\begin{tabular}[t]{l|l|l|l}
\hline
  & ID & UCSC\_REFGENE\_NAME & UCSC\_REFGENE\_GROUP\\
\hline
6 & cg00619207 & DENND2D & TSS200\\
\hline
7 & cg00705730 & NCK2;NCK2 & 5UTR;5UTR\\
\hline
10 & cg00921266 & HOXA3;HOXA3;HOXA3 & 5UTR;5UTR;TSS200\\
\hline
\end{tabular}

\textbackslash end\{table\}
\#\# Expression.Sample

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{knitr}\SpecialCharTok{::}\FunctionTok{kable}\NormalTok{(Expression.Sample[}\DecValTok{1}\SpecialCharTok{:}\DecValTok{5}\NormalTok{,}\DecValTok{1}\SpecialCharTok{:}\DecValTok{3}\NormalTok{], }\AttributeTok{caption =} \StringTok{\textquotesingle{}An R dataframe for the gene{-}centric gene expression (RNASeq) values of the tumor samples.\textquotesingle{}}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\begin{table}

\caption{\label{tab:unnamed-chunk-63}An R dataframe for the gene-centric gene expression (RNASeq) values of the tumor samples.}
\centering
\begin{tabular}[t]{l|r|r|r}
\hline
  & TCGA-OR-A5JT-01A & TCGA-OR-A5K6-01A & TCGA-OR-A5K3-01A\\
\hline
TSPAN6 & 10.0802526 & 13.3543602 & 10.9596161\\
\hline
TNMD & 0.2315967 & 0.0536225 & 0.0652658\\
\hline
DPM1 & 40.2218416 & 55.3076696 & 30.9633104\\
\hline
SCYL3 & 1.3136879 & 2.2388920 & 0.5801193\\
\hline
C1orf112 & 0.6405102 & 0.4268140 & 0.2524077\\
\hline
\end{tabular}
\end{table}

\hypertarget{expression.cellline}{%
\subsection{\texorpdfstring{ Expression.CellLine}{  Expression.CellLine}}\label{expression.cellline}}

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{knitr}\SpecialCharTok{::}\FunctionTok{kable}\NormalTok{(Expression.CellLine[}\DecValTok{1}\SpecialCharTok{:}\DecValTok{5}\NormalTok{,}\DecValTok{1}\SpecialCharTok{:}\DecValTok{3}\NormalTok{], }\AttributeTok{caption =} \StringTok{\textquotesingle{}An R dataframe for the gene{-}centric gene expression (RNASeq) values of the cell lines.\textquotesingle{}}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\begin{table}

\caption{\label{tab:unnamed-chunk-64}An R dataframe for the gene-centric gene expression (RNASeq) values of the cell lines.}
\centering
\begin{tabular}[t]{l|r|r|r}
\hline
  & LOUNH91\_LUNG & T98G\_CENTRAL\_NERVOUS\_SYSTEM & IPC298\_SKIN\\
\hline
TSPAN6 & 10.0802526 & 13.3543602 & 10.9596161\\
\hline
TNMD & 0.2315967 & 0.0536225 & 0.0652658\\
\hline
DPM1 & 40.2218416 & 55.3076696 & 30.9633104\\
\hline
SCYL3 & 1.3136879 & 2.2388920 & 0.5801193\\
\hline
C1orf112 & 0.6405102 & 0.4268140 & 0.2524077\\
\hline
\end{tabular}
\end{table}

\hypertarget{cancergenelist}{%
\subsection{\texorpdfstring{ CancerGeneList}{  CancerGeneList}}\label{cancergenelist}}

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{knitr}\SpecialCharTok{::}\FunctionTok{kable}\NormalTok{(CancerGeneList[}\DecValTok{1}\SpecialCharTok{:}\DecValTok{5}\NormalTok{,], }\AttributeTok{caption =} \StringTok{\textquotesingle{}An R dataframe object with a column of frequently mutated cancer driver genes.\textquotesingle{}}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\begin{table}

\caption{\label{tab:unnamed-chunk-65}An R dataframe object with a column of frequently mutated cancer driver genes.}
\centering
\begin{tabular}[t]{l}
\hline
V1\\
\hline
A1CF\\
\hline
ABI1\\
\hline
ABL1\\
\hline
ABL2\\
\hline
ACKR3\\
\hline
\end{tabular}
\end{table}

\hypertarget{genecentric.dnamethylation.sample}{%
\subsection{\texorpdfstring{ GeneCentric.DNAmethylation.Sample}{  GeneCentric.DNAmethylation.Sample}}\label{genecentric.dnamethylation.sample}}

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{knitr}\SpecialCharTok{::}\FunctionTok{kable}\NormalTok{(GeneCentric.DNAmethylation.Sample[}\DecValTok{1}\SpecialCharTok{:}\DecValTok{5}\NormalTok{,}\DecValTok{1}\SpecialCharTok{:}\DecValTok{3}\NormalTok{], }\AttributeTok{caption =} \StringTok{\textquotesingle{}An R dataframe with gene{-}centric DNA methylation beta values of bulk tumor samples. Rows are the genes, columns are the tumor samples.\textquotesingle{}}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\begin{table}

\caption{\label{tab:unnamed-chunk-66}An R dataframe with gene-centric DNA methylation beta values of bulk tumor samples. Rows are the genes, columns are the tumor samples.}
\centering
\begin{tabular}[t]{l|r|r|r}
\hline
  & TCGA-OU-A5PI-01A & TCGA-OR-A5K8-01A & TCGA-OR-A5JX-01A\\
\hline
A1BG & 0.5810136 & 0.2928524 & 0.5154303\\
\hline
A1CF & 0.7602958 & 0.5895629 & 0.7548407\\
\hline
A2BP1 & 0.6226549 & 0.5788986 & 0.3892227\\
\hline
A2LD1 & 0.9217585 & 0.9184202 & 0.9097963\\
\hline
A2M & 0.7231098 & 0.5392000 & 0.5754296\\
\hline
\end{tabular}
\end{table}

\hypertarget{genecentric.dnamethylation.cellline}{%
\subsection{\texorpdfstring{ GeneCentric.DNAmethylation.CellLine}{  GeneCentric.DNAmethylation.CellLine}}\label{genecentric.dnamethylation.cellline}}

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{knitr}\SpecialCharTok{::}\FunctionTok{kable}\NormalTok{(GeneCentric.DNAmethylation.CellLine[}\DecValTok{1}\SpecialCharTok{:}\DecValTok{5}\NormalTok{,}\DecValTok{1}\SpecialCharTok{:}\DecValTok{3}\NormalTok{], }\AttributeTok{caption =} \StringTok{\textquotesingle{}An R dataframe with gene{-}centric DNA methylation beta values of cell lines. Rows are the genes, columns are the cell lines.\textquotesingle{}}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\begin{table}

\caption{\label{tab:unnamed-chunk-67}An R dataframe with gene-centric DNA methylation beta values of cell lines. Rows are the genes, columns are the cell lines.}
\centering
\begin{tabular}[t]{l|r|r|r}
\hline
  & LOUNH91\_LUNG & T98G\_CENTRAL\_NERVOUS\_SYSTEM & IPC298\_SKIN\\
\hline
A1BG & 0.5810136 & 0.2928524 & 0.5154303\\
\hline
A1CF & 0.7602958 & 0.5895629 & 0.7548407\\
\hline
A2BP1 & 0.6226549 & 0.5788986 & 0.3892227\\
\hline
A2LD1 & 0.9217585 & 0.9184202 & 0.9097963\\
\hline
A2M & 0.7231098 & 0.5392000 & 0.5754296\\
\hline
\end{tabular}
\end{table}

\hypertarget{cnv.cellline}{%
\subsection{\texorpdfstring{ CNV.CellLine}{  CNV.CellLine}}\label{cnv.cellline}}

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{knitr}\SpecialCharTok{::}\FunctionTok{kable}\NormalTok{(CNV.CellLine[}\DecValTok{1}\SpecialCharTok{:}\DecValTok{5}\NormalTok{,}\DecValTok{1}\SpecialCharTok{:}\DecValTok{3}\NormalTok{], }\AttributeTok{caption =} \StringTok{\textquotesingle{}An R dataframe with gene{-}centric copy number values of cell lines. Rows are the genes, columns are the cell lines.\textquotesingle{}}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\begin{table}

\caption{\label{tab:unnamed-chunk-68}An R dataframe with gene-centric copy number values of cell lines. Rows are the genes, columns are the cell lines.}
\centering
\begin{tabular}[t]{l|r|r|r}
\hline
  & LOUNH91\_LUNG & T98G\_CENTRAL\_NERVOUS\_SYSTEM & IPC298\_SKIN\\
\hline
A1BG & 0.0259 & 0.1514 & 0.1511\\
\hline
NAT2 & 0.0325 & 0.0742 & 0.2180\\
\hline
ADA & 0.7455 & 0.5280 & 0.1834\\
\hline
CDH2 & 0.0136 & 0.4032 & 0.5538\\
\hline
AKT3 & 0.0488 & 0.1349 & 0.1940\\
\hline
\end{tabular}
\end{table}

\hypertarget{ctdpathsim2.0-functions}{%
\section{\texorpdfstring{ CTDPathsim2.0 Functions}{  CTDPathsim2.0 Functions}}\label{ctdpathsim2.0-functions}}

\hypertarget{step-1-computing-sample-specific-deconvoluted-methylation-profile.}{%
\subsection{\texorpdfstring{ Step 1: Computing sample-specific deconvoluted methylation profile.}{  Step 1: Computing sample-specific deconvoluted methylation profile.}}\label{step-1-computing-sample-specific-deconvoluted-methylation-profile.}}

\hypertarget{function1-rundeconvmethylfun}{%
\subsubsection{\texorpdfstring{ Function1: RunDeconvMethylFun}{  Function1: RunDeconvMethylFun}}\label{function1-rundeconvmethylfun}}

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{?RunDeconvMethylFun}
\DocumentationTok{\#\# i Rendering development documentation for \textquotesingle{}RunDeconvMethylFun\textquotesingle{}}
\end{Highlighting}
\end{Shaded}

\begin{Shaded}
\begin{Highlighting}[]
\CommentTok{\# Running the function}
\NormalTok{stage1\_result\_ct }\OtherTok{\textless{}{-}}
  \FunctionTok{RunDeconvMethylFun}\NormalTok{(Methylation.Sample,Reference.Methylation.Markers)}
\end{Highlighting}
\end{Shaded}

\begin{Shaded}
\begin{Highlighting}[]
\CommentTok{\# Outputs}
\NormalTok{knitr}\SpecialCharTok{::}\FunctionTok{kable}\NormalTok{(stage1\_result\_ct}\SpecialCharTok{$}\NormalTok{methylation[}\DecValTok{1}\SpecialCharTok{:}\DecValTok{5}\NormalTok{,], }\AttributeTok{caption =} \StringTok{\textquotesingle{}Estimated DNA methylation profile of constituent cell types.\textquotesingle{}}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\begin{table}

\caption{\label{tab:unnamed-chunk-71}Estimated DNA methylation profile of constituent cell types.}
\centering
\begin{tabular}[t]{l|r|r|r}
\hline
cg00323915 & 0.1829631 & 0.1606620 & 0.5224480\\
\hline
cg21830221 & 0.6576252 & 0.8006095 & 0.9129288\\
\hline
cg11348106 & 0.7750180 & 0.6967344 & 0.8727026\\
\hline
cg12232731 & 0.5502089 & 0.5240266 & 0.6487635\\
\hline
cg07218880 & 0.3673941 & 0.0751428 & 0.5104521\\
\hline
\end{tabular}
\end{table}

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{knitr}\SpecialCharTok{::}\FunctionTok{kable}\NormalTok{(stage1\_result\_ct}\SpecialCharTok{$}\NormalTok{proportions[}\DecValTok{1}\SpecialCharTok{:}\DecValTok{5}\NormalTok{,], }\AttributeTok{caption =} \StringTok{\textquotesingle{}Estimated proportions of constituent cell types.\textquotesingle{}}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\begin{table}

\caption{\label{tab:unnamed-chunk-71}Estimated proportions of constituent cell types.}
\centering
\begin{tabular}[t]{l|r|r|r}
\hline
TCGA-OU-A5PI-01A & 0.00 & 0.31 & 0.69\\
\hline
TCGA-OR-A5K8-01A & 0.00 & 0.55 & 0.45\\
\hline
TCGA-OR-A5JX-01A & 0.07 & 0.86 & 0.07\\
\hline
TCGA-OR-A5LT-01A & 0.23 & 0.58 & 0.19\\
\hline
TCGA-PK-A5H9-01A & 0.20 & 0.73 & 0.07\\
\hline
\end{tabular}
\end{table}

\hypertarget{function2-plotdeconvmethylfun}{%
\subsubsection{\texorpdfstring{ Function2: PlotDeconvMethylFun}{  Function2: PlotDeconvMethylFun}}\label{function2-plotdeconvmethylfun}}

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{?PlotDeconvMethylFun}
\DocumentationTok{\#\# i Rendering development documentation for \textquotesingle{}PlotDeconvMethylFun\textquotesingle{}}
\end{Highlighting}
\end{Shaded}

\begin{Shaded}
\begin{Highlighting}[]
\CommentTok{\# Running the function}
\FunctionTok{print}\NormalTok{(}
\FunctionTok{PlotDeconvMethylFun}\NormalTok{(Methylation.Sample,Reference.Methylation.Markers,Reference.Methylation.CellTypes,stage1\_result\_ct,Reference.CellTypes.Names)}
\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\begin{adjustwidth}{\fltoffset}{0mm}
\includegraphics[width=1\linewidth,]{CTDPathSim2.0_files/figure-latex/unnamed-chunk-73-1} \end{adjustwidth}

\begin{verbatim}
## $rowInd
##  [1]  7 28 30 19 21 11 12 23  4 17 20  8 27  5 26 24  6 31  2  9  1 14 22 29  3 13
## [27] 10 25 16 18 15 32 33 37 38 36 34 35
## 
## $colInd
## [1] 2 3 1
## 
## $call
## gplots::heatmap.2(x = cors_deconv_refs_ct, breaks = seq(0, 1, 
##     0.1), col = color_gradient(10), cellnote = best_cor_labels, 
##     notecol = "black", trace = "none", margins = c(5, 5), RowSideColors = ref_class_colors)
## 
## $carpet
##           CD4T    CD4T_5   CD4T_6    CD4T_3    CD4T_4    CD4T_1    CD4T_2
## [1,] 0.0000000 0.0000000 0.000000 0.0000000 0.0000000 0.0000000 0.0000000
## [2,] 0.0000000 0.0000000 0.000000 0.0000000 0.0000000 0.0000000 0.0000000
## [3,] 0.2397563 0.2398837 0.239822 0.2356866 0.2318461 0.2490854 0.2706678
##       Bcell_4     Bcell  Bcell_2   Bcell_3   Bcell_1   Bcell_5      NK_1
## [1,] 0.000000 0.0000000 0.000000 0.0000000 0.0000000 0.0000000 0.0000000
## [2,] 0.000000 0.0000000 0.000000 0.0000000 0.0000000 0.0000000 0.0000000
## [3,] 0.214543 0.2134951 0.209603 0.2038772 0.2077764 0.1947861 0.3078642
##           NK_4      NK_3      NK_2     NK_5        NK     Neu_2      Neu
## [1,] 0.0000000 0.0000000 0.0000000 0.000000 0.0000000 0.0000000 0.000000
## [2,] 0.0000000 0.0000000 0.0000000 0.000000 0.0000000 0.0000000 0.000000
## [3,] 0.2873833 0.2901649 0.2977637 0.296879 0.3038225 0.3504125 0.353996
##          Neu_3     Neu_4     Neu_5     Neu_1       Mono_1      Mono    Mono_5
## [1,] 0.0000000 0.0000000 0.0000000 0.0000000 7.686684e-06 0.0000000 0.0000000
## [2,] 0.0000000 0.0000000 0.0000000 0.0000000 0.000000e+00 0.0000000 0.0000000
## [3,] 0.3543978 0.3596619 0.3568697 0.3614782 4.682099e-01 0.4500065 0.4544081
##         Mono_3    Mono_4    Mono_2 cortical_neurons_rep2 cortical_neurons_rep3
## [1,] 0.0000000 0.0000000 0.0000000             0.6594859             0.6693292
## [2,] 0.0000000 0.0000000 0.0000000             0.4877121             0.5047259
## [3,] 0.4514482 0.4562684 0.4599231             0.2634952             0.2799626
##      adipocytes_rep2 adipocytes_rep3 adipocytes_rep1
## [1,]       0.6732395       0.6731809       0.6855557
## [2,]       0.5325775       0.5374546       0.5485267
## [3,]       0.3983162       0.4063788       0.4621006
##      vascular_endothelial_cells_rep1 vascular_endothelial_cells_rep2
## [1,]                       0.7220795                       0.7299448
## [2,]                       0.5682841                       0.5753553
## [3,]                       0.4304075                       0.4423302
## 
## $rowDendrogram
## 'dendrogram' with 2 branches and 38 members total, at height 1.196426 
## 
## $colDendrogram
## 'dendrogram' with 2 branches and 3 members total, at height 2.550261 
## 
## $breaks
##  [1] 0.0 0.1 0.2 0.3 0.4 0.5 0.6 0.7 0.8 0.9 1.0
## 
## $col
##  [1] "#FFFFFF" "#EAF1F6" "#D5E3EE" "#C1D5E6" "#ACC7DD" "#98B9D5" "#83ABCD"
##  [8] "#6F9DC4" "#5A8FBC" "#4682B4"
## 
## $colorTable
##    low high   color
## 1  0.0  0.1 #FFFFFF
## 2  0.1  0.2 #EAF1F6
## 3  0.2  0.3 #D5E3EE
## 4  0.3  0.4 #C1D5E6
## 5  0.4  0.5 #ACC7DD
## 6  0.5  0.6 #98B9D5
## 7  0.6  0.7 #83ABCD
## 8  0.7  0.8 #6F9DC4
## 9  0.8  0.9 #5A8FBC
## 10 0.9  1.0 #4682B4
## 
## $layout
## $layout$lmat
##      [,1] [,2] [,3]
## [1,]    5    0    4
## [2,]    3    1    2
## 
## $layout$lhei
## [1] 1.5 4.0
## 
## $layout$lwid
## [1] 1.5 0.2 4.0
dev.off()
## null device 
##           1
\end{verbatim}

\hypertarget{function3-probetogenefun}{%
\subsubsection{\texorpdfstring{ Function3: ProbeToGeneFun}{  Function3: ProbeToGeneFun}}\label{function3-probetogenefun}}

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{?ProbeToGeneFun}
\DocumentationTok{\#\# i Rendering development documentation for \textquotesingle{}ProbeToGeneFun\textquotesingle{}}
\CommentTok{\# Running the function}
\NormalTok{Deconv\_methylation }\OtherTok{\textless{}{-}}\NormalTok{ stage1\_result\_ct}\SpecialCharTok{$}\NormalTok{methylation}
\NormalTok{Deconv\_meth\_gene }\OtherTok{\textless{}{-}}
  \FunctionTok{ProbeToGeneFun}\NormalTok{(Annotation450K,Deconv\_methylation,}\StringTok{"450K"}\NormalTok{)}
\CommentTok{\# Output}
\NormalTok{knitr}\SpecialCharTok{::}\FunctionTok{kable}\NormalTok{(Deconv\_meth\_gene[}\DecValTok{1}\SpecialCharTok{:}\DecValTok{5}\NormalTok{,], }\AttributeTok{caption =} \StringTok{\textquotesingle{}An R dataframe object with deconvoluted DNA methylation values for each gene in cell types. The rows are the genes and the columns are the cell types. This function labels the columns as V1, V2, V3,....etc.\textquotesingle{}}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\begin{table}

\caption{\label{tab:unnamed-chunk-74}An R dataframe object with deconvoluted DNA methylation values for each gene in cell types. The rows are the genes and the columns are the cell types. This function labels the columns as V1, V2, V3,....etc.}
\centering
\begin{tabular}[t]{l|r|r|r}
\hline
  & V1 & V2 & V3\\
\hline
ABI3 & 0.8692453 & 0.9318306 & 0.9343205\\
\hline
ADORA2A & 0.4833491 & 0.9693039 & 0.9688007\\
\hline
ANKRD11 & 0.3610541 & 0.8464815 & 0.9703335\\
\hline
BIN2 & 0.5049618 & 0.9823414 & 0.9621162\\
\hline
BLK & 0.2800716 & 0.5299947 & 0.8206108\\
\hline
\end{tabular}
\end{table}

\hypertarget{function4-samplemethylfun}{%
\subsubsection{\texorpdfstring{ Function4: SampleMethylFun}{  Function4: SampleMethylFun}}\label{function4-samplemethylfun}}

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{?SampleMethylFun}
\DocumentationTok{\#\# i Rendering development documentation for \textquotesingle{}SampleMethylFun\textquotesingle{}}
\CommentTok{\# Running the function}
\NormalTok{CTDDirectory }\OtherTok{\textless{}{-}} \FunctionTok{tempdir}\NormalTok{()}
\FunctionTok{SampleMethylFun}\NormalTok{(Deconv\_meth\_gene,Deconv\_proportions, CTDDirectory)}
\DocumentationTok{\#\# NULL}
\CommentTok{\# Output}
\FunctionTok{load}\NormalTok{(}\StringTok{"\textasciitilde{}/Sample\_methylation/TCGA{-}OR{-}A5J1{-}01A.Rda"}\NormalTok{)}
\NormalTok{knitr}\SpecialCharTok{::}\FunctionTok{kable}\NormalTok{(methylation[}\DecValTok{1}\SpecialCharTok{:}\DecValTok{5}\NormalTok{,], }\AttributeTok{caption =} \StringTok{\textquotesingle{}Deconvoluted DNA methylation profile of the sample TCGA{-}OR{-}A5J1{-}01A\textquotesingle{}}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\begin{table}

\caption{\label{tab:unnamed-chunk-75}Deconvoluted DNA methylation profile of the sample TCGA-OR-A5J1-01A}
\centering
\begin{tabular}[t]{l|r|r|r}
\hline
  & V1 & V2 & V3\\
\hline
ABI3 & 0.3075042 & 0.0086925 & 0.6166515\\
\hline
ADORA2A & 0.3198708 & 0.0048335 & 0.6394082\\
\hline
ANKRD11 & 0.2793398 & 0.0036106 & 0.6404200\\
\hline
BIN2 & 0.3241731 & 0.0050497 & 0.6349964\\
\hline
BLK & 0.1748998 & 0.0028007 & 0.5416030\\
\hline
\end{tabular}
\end{table}

\hypertarget{step-2-computing-sample-specific-deconvoluted-expression-profile.}{%
\subsection{\texorpdfstring{ Step 2: Computing sample-specific deconvoluted expression profile.}{  Step 2: Computing sample-specific deconvoluted expression profile.}}\label{step-2-computing-sample-specific-deconvoluted-expression-profile.}}

\hypertarget{function1-rundeconvexprfun}{%
\subsubsection{\texorpdfstring{ Function1: RunDeconvExprFun}{  Function1: RunDeconvExprFun}}\label{function1-rundeconvexprfun}}

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{?RunDeconvExprFun}
\DocumentationTok{\#\# i Rendering development documentation for \textquotesingle{}RunDeconvExprFun\textquotesingle{}}
\CommentTok{\# Running the function}
\NormalTok{Deconv\_expression}\OtherTok{\textless{}{-}}\FunctionTok{RunDeconvExprFun}\NormalTok{(Expression.Sample,Deconv\_proportions)}
\CommentTok{\# Output}
\NormalTok{knitr}\SpecialCharTok{::}\FunctionTok{kable}\NormalTok{(Deconv\_expression[}\DecValTok{1}\SpecialCharTok{:}\DecValTok{5}\NormalTok{,], }\AttributeTok{caption =} \StringTok{\textquotesingle{}An R dataframe object with deconvoluted gene expression values for each gene in cell types. The rows are the genes and the columns are the cell types. This function labels the columns as V1, V2, V3,....etc.\textquotesingle{}}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\begin{table}

\caption{\label{tab:unnamed-chunk-76}An R dataframe object with deconvoluted gene expression values for each gene in cell types. The rows are the genes and the columns are the cell types. This function labels the columns as V1, V2, V3,....etc.}
\centering
\begin{tabular}[t]{l|r|r|r}
\hline
TSPAN6 & 10.2170293 & 7.662672 & 7.4049619\\
\hline
TNMD & 0.1067874 & 0.244172 & 0.0321168\\
\hline
DPM1 & 35.6648144 & 25.368440 & 42.1562712\\
\hline
SCYL3 & 1.3917714 & 1.781208 & 1.2404976\\
\hline
C1orf112 & 0.2324402 & 1.249619 & 0.9056265\\
\hline
\end{tabular}
\end{table}

\hypertarget{function2-sampleexprfun}{%
\subsubsection{\texorpdfstring{ Function2: SampleExprFun}{  Function2: SampleExprFun}}\label{function2-sampleexprfun}}

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{?SampleExprFun}
\DocumentationTok{\#\# i Rendering development documentation for \textquotesingle{}SampleExprFun\textquotesingle{}}
\CommentTok{\# Running the function}
\FunctionTok{SampleExprFun}\NormalTok{(Deconv\_expression,Deconv\_proportions, CTDDirectory)}
\DocumentationTok{\#\# NULL}
\CommentTok{\# Output}
\FunctionTok{load}\NormalTok{(}\StringTok{"\textasciitilde{}/Sample\_expression/TCGA{-}OR{-}A5J1{-}01A.Rda"}\NormalTok{)}
\NormalTok{knitr}\SpecialCharTok{::}\FunctionTok{kable}\NormalTok{(expression[}\DecValTok{1}\SpecialCharTok{:}\DecValTok{5}\NormalTok{,], }\AttributeTok{caption =} \StringTok{\textquotesingle{}Deconvoluted gene expression profile of the sample TCGA{-}OR{-}A5J1{-}01A\textquotesingle{}}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\begin{table}

\caption{\label{tab:unnamed-chunk-77}Deconvoluted gene expression profile of the sample TCGA-OR-A5J1-01A}
\centering
\begin{tabular}[t]{l|r|r|r}
\hline
TSPAN6 & 3.3716197 & 0.0766267 & 4.8872748\\
\hline
TNMD & 0.0352398 & 0.0024417 & 0.0211971\\
\hline
DPM1 & 11.7693888 & 0.2536844 & 27.8231390\\
\hline
SCYL3 & 0.4592846 & 0.0178121 & 0.8187284\\
\hline
C1orf112 & 0.0767053 & 0.0124962 & 0.5977135\\
\hline
\end{tabular}
\end{table}

\hypertarget{step-3-computing-sample-and-cell-line-specific-differentially-expressed-de-genes-and-enriched-biological-pathways.}{%
\subsection{\texorpdfstring{ Step 3: Computing sample and cell line-specific differentially expressed (DE) genes and enriched biological pathways.}{  Step 3: Computing sample and cell line-specific differentially expressed (DE) genes and enriched biological pathways.}}\label{step-3-computing-sample-and-cell-line-specific-differentially-expressed-de-genes-and-enriched-biological-pathways.}}

\hypertarget{function1-getsampledefun}{%
\subsubsection{\texorpdfstring{ Function1: GetSampleDEFun}{  Function1: GetSampleDEFun}}\label{function1-getsampledefun}}

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{?GetSampleDEFun}
\DocumentationTok{\#\# i Rendering development documentation for \textquotesingle{}GetSampleDEFun\textquotesingle{}}
\CommentTok{\# Running the function}
\NormalTok{CTDDirectory }\OtherTok{\textless{}{-}} \FunctionTok{tempdir}\NormalTok{()}
\FunctionTok{GetSampleDEFun}\NormalTok{(}\AttributeTok{RnaSeq\_data=}\NormalTok{Expression.Sample,}\AttributeTok{parallel=} \ConstantTok{TRUE}\NormalTok{,}\AttributeTok{ncores=}\DecValTok{2}\NormalTok{, CTDDirectory)}
\DocumentationTok{\#\# Warning in dir.create(paste0(CTDDirectory, "/Patient\_DE\_genes/")): \textquotesingle{}C:}
\DocumentationTok{\#\# \textbackslash{}Users\textbackslash{}Sarah\textbackslash{}AppData\textbackslash{}Local\textbackslash{}Temp\textbackslash{}Rtmp4q5ue2\textbackslash{}Patient\_DE\_genes\textquotesingle{} already exists}
\DocumentationTok{\#\# Warning in dir.create(paste0(CTDDirectory, "/Patient\_DE\_genes/UP\_Gene/")): \textquotesingle{}C:}
\DocumentationTok{\#\# \textbackslash{}Users\textbackslash{}Sarah\textbackslash{}AppData\textbackslash{}Local\textbackslash{}Temp\textbackslash{}Rtmp4q5ue2\textbackslash{}Patient\_DE\_genes\textbackslash{}UP\_Gene\textquotesingle{} already}
\DocumentationTok{\#\# exists}
\DocumentationTok{\#\# Warning in dir.create(paste0(CTDDirectory, "/Patient\_DE\_genes/DOWN\_Gene/")): \textquotesingle{}C:}
\DocumentationTok{\#\# \textbackslash{}Users\textbackslash{}Sarah\textbackslash{}AppData\textbackslash{}Local\textbackslash{}Temp\textbackslash{}Rtmp4q5ue2\textbackslash{}Patient\_DE\_genes\textbackslash{}DOWN\_Gene\textquotesingle{} already}
\DocumentationTok{\#\# exists}
\DocumentationTok{\#\# Warning in dir.create(paste0(CTDDirectory, "/Patient\_DE\_genes/DE\_Gene/")): \textquotesingle{}C:}
\DocumentationTok{\#\# \textbackslash{}Users\textbackslash{}Sarah\textbackslash{}AppData\textbackslash{}Local\textbackslash{}Temp\textbackslash{}Rtmp4q5ue2\textbackslash{}Patient\_DE\_genes\textbackslash{}DE\_Gene\textquotesingle{} already}
\DocumentationTok{\#\# exists}
\DocumentationTok{\#\# [1] "cluster making started"}
\DocumentationTok{\#\# [1] "cluster export started"}
\DocumentationTok{\#\# [1] "cluster making done"}
\DocumentationTok{\#\# [1] "cluster stopped"}
\DocumentationTok{\#\# NULL}
\CommentTok{\# Output}
\FunctionTok{load}\NormalTok{(}\FunctionTok{paste0}\NormalTok{(CTDDirectory,}\StringTok{"/Patient\_DE\_genes/DE\_Gene/TCGA{-}OR{-}A5J6{-}01A\_pat\_de\_genes.Rda"}\NormalTok{))}
\NormalTok{knitr}\SpecialCharTok{::}\FunctionTok{kable}\NormalTok{(pat\_de\_genes, }\AttributeTok{caption =} \StringTok{\textquotesingle{}DE genes of the sample TCGA{-}OR{-}A5J1{-}01A\textquotesingle{}}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\begin{table}

\caption{\label{tab:unnamed-chunk-78}DE genes of the sample TCGA-OR-A5J1-01A}
\centering
\begin{tabular}[t]{l}
\hline
gene\\
\hline
TNMD\\
\hline
\end{tabular}
\end{table}

\hypertarget{function2-getcelllinedefun}{%
\subsubsection{\texorpdfstring{ Function2: GetCellLineDEFun}{  Function2: GetCellLineDEFun}}\label{function2-getcelllinedefun}}

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{?GetCellLineDEFun}
\DocumentationTok{\#\# i Rendering development documentation for \textquotesingle{}GetCellLineDEFun\textquotesingle{}}
\CommentTok{\# Running the function}
\FunctionTok{GetCellLineDEFun}\NormalTok{(}\AttributeTok{RnaSeq\_data=}\NormalTok{Expression.CellLine,}\AttributeTok{parallel=} \ConstantTok{TRUE}\NormalTok{,}\AttributeTok{ncores=}\DecValTok{2}\NormalTok{, CTDDirectory)}
\DocumentationTok{\#\# Warning in dir.create(paste0(CTDDirectory, "/CellLine\_DE\_genes/")): \textquotesingle{}C:}
\DocumentationTok{\#\# \textbackslash{}Users\textbackslash{}Sarah\textbackslash{}AppData\textbackslash{}Local\textbackslash{}Temp\textbackslash{}Rtmp4q5ue2\textbackslash{}CellLine\_DE\_genes\textquotesingle{} already exists}
\DocumentationTok{\#\# Warning in dir.create(paste0(CTDDirectory, "/CellLine\_DE\_genes/UP\_Gene/")): \textquotesingle{}C:}
\DocumentationTok{\#\# \textbackslash{}Users\textbackslash{}Sarah\textbackslash{}AppData\textbackslash{}Local\textbackslash{}Temp\textbackslash{}Rtmp4q5ue2\textbackslash{}CellLine\_DE\_genes\textbackslash{}UP\_Gene\textquotesingle{} already}
\DocumentationTok{\#\# exists}
\DocumentationTok{\#\# Warning in dir.create(paste0(CTDDirectory, "/CellLine\_DE\_genes/DOWN\_Gene/")): \textquotesingle{}C:}
\DocumentationTok{\#\# \textbackslash{}Users\textbackslash{}Sarah\textbackslash{}AppData\textbackslash{}Local\textbackslash{}Temp\textbackslash{}Rtmp4q5ue2\textbackslash{}CellLine\_DE\_genes\textbackslash{}DOWN\_Gene\textquotesingle{} already}
\DocumentationTok{\#\# exists}
\DocumentationTok{\#\# Warning in dir.create(paste0(CTDDirectory, "/CellLine\_DE\_genes/DE\_Gene/")): \textquotesingle{}C:}
\DocumentationTok{\#\# \textbackslash{}Users\textbackslash{}Sarah\textbackslash{}AppData\textbackslash{}Local\textbackslash{}Temp\textbackslash{}Rtmp4q5ue2\textbackslash{}CellLine\_DE\_genes\textbackslash{}DE\_Gene\textquotesingle{} already}
\DocumentationTok{\#\# exists}
\DocumentationTok{\#\# [1] "cluster making started"}
\DocumentationTok{\#\# [1] "cluster export started"}
\DocumentationTok{\#\# [1] "cluster making done"}
\DocumentationTok{\#\# [1] "cluster stopped"}
\DocumentationTok{\#\# NULL}
\CommentTok{\# Output}
\FunctionTok{load}\NormalTok{(}\FunctionTok{paste0}\NormalTok{(CTDDirectory,}\StringTok{"/CellLine\_DE\_genes/DE\_Gene/LK2\_LUNG\_cell\_de\_genes.Rda"}\NormalTok{))}
\NormalTok{knitr}\SpecialCharTok{::}\FunctionTok{kable}\NormalTok{(cell\_de\_genes[}\DecValTok{1}\SpecialCharTok{:}\DecValTok{5}\NormalTok{,,}\AttributeTok{drop=}\NormalTok{F], }\AttributeTok{caption =} \StringTok{\textquotesingle{}DE genes of the LK2 lung cancer cell line\textquotesingle{}}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\begin{table}

\caption{\label{tab:unnamed-chunk-79}DE genes of the LK2 lung cancer cell line}
\centering
\begin{tabular}[t]{l}
\hline
gene\\
\hline
A1CF\\
\hline
A2M\\
\hline
A2M-AS1\\
\hline
A2ML1-AS1\\
\hline
A2MP1\\
\hline
\end{tabular}
\end{table}

\hypertarget{function3-getpathfun}{%
\subsubsection{\texorpdfstring{ Function3: GetPathFun}{  Function3: GetPathFun}}\label{function3-getpathfun}}

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{?GetPathFun}
\DocumentationTok{\#\# i Rendering development documentation for \textquotesingle{}GetPathFun\textquotesingle{}}
\CommentTok{\# Running the function}
\FunctionTok{data}\NormalTok{(}\StringTok{"CTDPathSim2"}\NormalTok{)}
\NormalTok{Enriched.pathways.sample}\OtherTok{\textless{}{-}}\FunctionTok{GetPathFun}\NormalTok{(CancerGeneList,pat\_de\_genes)}
\NormalTok{Enriched.pathways.cellLine}\OtherTok{\textless{}{-}}\FunctionTok{GetPathFun}\NormalTok{(CancerGeneList,cell\_de\_genes)}
\CommentTok{\# Output}
\NormalTok{knitr}\SpecialCharTok{::}\FunctionTok{kable}\NormalTok{(Enriched.pathways.sample[}\DecValTok{1}\SpecialCharTok{:}\DecValTok{5}\NormalTok{,], }\AttributeTok{caption =} \StringTok{\textquotesingle{}Sample(TCGA{-}OR{-}A5J1{-}01A){-}specific enriched biological pathways\textquotesingle{}}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\begin{table}

\caption{\label{tab:unnamed-chunk-80}Sample(TCGA-OR-A5J1-01A)-specific enriched biological pathways}
\centering
\begin{tabular}[t]{l|l|r}
\hline
ID & reactome\_pathway & p.adjust.pat\\
\hline
R-HSA-162582 & Signal Transduction & 0.00e+00\\
\hline
R-HSA-1266738 & Developmental Biology & 0.00e+00\\
\hline
R-HSA-1643685 & Disease & 0.00e+00\\
\hline
R-HSA-212436 & Generic Transcription Pathway & 4.60e-06\\
\hline
R-HSA-73857 & RNA Polymerase II Transcription & 1.28e-05\\
\hline
\end{tabular}
\end{table}

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{knitr}\SpecialCharTok{::}\FunctionTok{kable}\NormalTok{(Enriched.pathways.cellLine[}\DecValTok{1}\SpecialCharTok{:}\DecValTok{5}\NormalTok{,], }\AttributeTok{caption =} \StringTok{\textquotesingle{}Cell line(LK2){-}specific enriched biological pathways\textquotesingle{}}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\begin{table}

\caption{\label{tab:unnamed-chunk-80}Cell line(LK2)-specific enriched biological pathways}
\centering
\begin{tabular}[t]{l|l|r}
\hline
ID & reactome\_pathway & p.adjust.pat\\
\hline
R-HSA-162582 & Signal Transduction & 0\\
\hline
R-HSA-168256 & Immune System & 0\\
\hline
R-HSA-9006934 & Signaling by Receptor Tyrosine Kinases & 0\\
\hline
R-HSA-1280215 & Cytokine Signaling in Immune system & 0\\
\hline
R-HSA-5663202 & Diseases of signal transduction by growth factor receptors and second messengers & 0\\
\hline
\end{tabular}
\end{table}

\hypertarget{step-4-computing-sample-and-cell-line-specific-differentially-methylated-dm-and-differentially-aberrated-da-genes.}{%
\subsection{\texorpdfstring{ Step 4: Computing sample and cell line-specific differentially methylated (DM) and differentially aberrated (DA) genes.}{  Step 4: Computing sample and cell line-specific differentially methylated (DM) and differentially aberrated (DA) genes.}}\label{step-4-computing-sample-and-cell-line-specific-differentially-methylated-dm-and-differentially-aberrated-da-genes.}}

\hypertarget{function1-getsampledmfun}{%
\subsubsection{\texorpdfstring{ Function1: GetSampleDMFun}{  Function1: GetSampleDMFun}}\label{function1-getsampledmfun}}

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{?GetSampleDMFun}
\DocumentationTok{\#\# i Rendering development documentation for \textquotesingle{}GetSampleDMFun\textquotesingle{}}
\CommentTok{\# Running the function}
\FunctionTok{GetSampleDMFun}\NormalTok{(GeneCentric.DNAmethylation.Sample,}\AttributeTok{parallel=} \ConstantTok{TRUE}\NormalTok{,}\AttributeTok{ncores=}\DecValTok{2}\NormalTok{, CTDDirectory)}
\DocumentationTok{\#\# Warning in dir.create(paste0(CTDDirectory, "/Patient\_DM\_genes/")): \textquotesingle{}C:}
\DocumentationTok{\#\# \textbackslash{}Users\textbackslash{}Sarah\textbackslash{}AppData\textbackslash{}Local\textbackslash{}Temp\textbackslash{}Rtmp4q5ue2\textbackslash{}Patient\_DM\_genes\textquotesingle{} already exists}
\DocumentationTok{\#\# Warning in dir.create(paste0(CTDDirectory, "/Patient\_DM\_genes/UP\_Gene/")): \textquotesingle{}C:}
\DocumentationTok{\#\# \textbackslash{}Users\textbackslash{}Sarah\textbackslash{}AppData\textbackslash{}Local\textbackslash{}Temp\textbackslash{}Rtmp4q5ue2\textbackslash{}Patient\_DM\_genes\textbackslash{}UP\_Gene\textquotesingle{} already}
\DocumentationTok{\#\# exists}
\DocumentationTok{\#\# Warning in dir.create(paste0(CTDDirectory, "/Patient\_DM\_genes/DOWN\_Gene/")): \textquotesingle{}C:}
\DocumentationTok{\#\# \textbackslash{}Users\textbackslash{}Sarah\textbackslash{}AppData\textbackslash{}Local\textbackslash{}Temp\textbackslash{}Rtmp4q5ue2\textbackslash{}Patient\_DM\_genes\textbackslash{}DOWN\_Gene\textquotesingle{} already}
\DocumentationTok{\#\# exists}
\DocumentationTok{\#\# Warning in dir.create(paste0(CTDDirectory, "/Patient\_DM\_genes/DM\_Gene/")): \textquotesingle{}C:}
\DocumentationTok{\#\# \textbackslash{}Users\textbackslash{}Sarah\textbackslash{}AppData\textbackslash{}Local\textbackslash{}Temp\textbackslash{}Rtmp4q5ue2\textbackslash{}Patient\_DM\_genes\textbackslash{}DM\_Gene\textquotesingle{} already}
\DocumentationTok{\#\# exists}
\DocumentationTok{\#\# [1] "cluster making started"}
\DocumentationTok{\#\# [1] "cluster export started"}
\DocumentationTok{\#\# [1] "cluster making done"}
\DocumentationTok{\#\# [1] "cluster stopped"}
\DocumentationTok{\#\# NULL}
\CommentTok{\# Output}
\FunctionTok{load}\NormalTok{(}\FunctionTok{paste0}\NormalTok{(CTDDirectory,}\StringTok{"/Patient\_DM\_genes/DM\_Gene/TCGA{-}OR{-}A5J6{-}01A\_pat\_dm\_genes.Rda"}\NormalTok{))}
\NormalTok{knitr}\SpecialCharTok{::}\FunctionTok{kable}\NormalTok{(pat\_dm\_genes, }\AttributeTok{caption =} \StringTok{\textquotesingle{}DM genes of the sample TCGA{-}OR{-}A5J1{-}01A\textquotesingle{}}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\begin{table}

\caption{\label{tab:unnamed-chunk-81}DM genes of the sample TCGA-OR-A5J1-01A}
\centering
\begin{tabular}[t]{l}
\hline
gene\\
\hline
A1CF\\
\hline
A2M\\
\hline
A4GALT\\
\hline
ABAT\\
\hline
ABCA1\\
\hline
AADACL2\\
\hline
ABCA10\\
\hline
\end{tabular}
\end{table}

\hypertarget{function2-getcelllinedmfun}{%
\subsubsection{\texorpdfstring{ Function2: GetCellLineDMFun}{  Function2: GetCellLineDMFun}}\label{function2-getcelllinedmfun}}

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{?GetCellLineDMFun}
\DocumentationTok{\#\# i Rendering development documentation for \textquotesingle{}GetCellLineDMFun\textquotesingle{}}
\CommentTok{\# Running the function}
\FunctionTok{GetCellLineDMFun}\NormalTok{(GeneCentric.DNAmethylation.CellLine,}\AttributeTok{parallel=} \ConstantTok{TRUE}\NormalTok{,}\AttributeTok{ncores=}\DecValTok{2}\NormalTok{, CTDDirectory)}
\DocumentationTok{\#\# Warning in dir.create(paste0(CTDDirectory, "/CellLine\_DM\_genes/")): \textquotesingle{}C:}
\DocumentationTok{\#\# \textbackslash{}Users\textbackslash{}Sarah\textbackslash{}AppData\textbackslash{}Local\textbackslash{}Temp\textbackslash{}Rtmp4q5ue2\textbackslash{}CellLine\_DM\_genes\textquotesingle{} already exists}
\DocumentationTok{\#\# Warning in dir.create(paste0(CTDDirectory, "/CellLine\_DM\_genes/UP\_Gene/")): \textquotesingle{}C:}
\DocumentationTok{\#\# \textbackslash{}Users\textbackslash{}Sarah\textbackslash{}AppData\textbackslash{}Local\textbackslash{}Temp\textbackslash{}Rtmp4q5ue2\textbackslash{}CellLine\_DM\_genes\textbackslash{}UP\_Gene\textquotesingle{} already}
\DocumentationTok{\#\# exists}
\DocumentationTok{\#\# Warning in dir.create(paste0(CTDDirectory, "/CellLine\_DM\_genes/DOWN\_Gene/")): \textquotesingle{}C:}
\DocumentationTok{\#\# \textbackslash{}Users\textbackslash{}Sarah\textbackslash{}AppData\textbackslash{}Local\textbackslash{}Temp\textbackslash{}Rtmp4q5ue2\textbackslash{}CellLine\_DM\_genes\textbackslash{}DOWN\_Gene\textquotesingle{} already}
\DocumentationTok{\#\# exists}
\DocumentationTok{\#\# [1] "cluster making started"}
\DocumentationTok{\#\# [1] "cluster export started"}
\DocumentationTok{\#\# [1] "cluster making done"}
\DocumentationTok{\#\# [1] "cluster stopped"}
\DocumentationTok{\#\# NULL}
\CommentTok{\# Output}
\FunctionTok{load}\NormalTok{(}\FunctionTok{paste0}\NormalTok{(CTDDirectory,}\StringTok{"/CellLine\_DM\_genes/DM\_Gene/LK2\_LUNG\_cell\_dm\_genes.Rda"}\NormalTok{))}
\NormalTok{knitr}\SpecialCharTok{::}\FunctionTok{kable}\NormalTok{(cell\_dm\_genes[}\DecValTok{1}\SpecialCharTok{:}\DecValTok{5}\NormalTok{,], }\AttributeTok{caption =} \StringTok{\textquotesingle{}DM genes of the LK2 cell line\textquotesingle{}}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\begin{table}

\caption{\label{tab:unnamed-chunk-82}DM genes of the LK2 cell line}
\centering
\begin{tabular}[t]{l}
\hline
gene\\
\hline
ADCY9\\
\hline
ADCYAP1\\
\hline
ADCYAP1R1\\
\hline
ADD3\\
\hline
ADGRE1\\
\hline
\end{tabular}
\end{table}

\hypertarget{function3-getcelllinedafun}{%
\subsubsection{\texorpdfstring{ Function3: GetCellLineDAFun}{  Function3: GetCellLineDAFun}}\label{function3-getcelllinedafun}}

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{?GetCellLineDAFun}
\DocumentationTok{\#\# i Rendering development documentation for \textquotesingle{}GetCellLineDAFun\textquotesingle{}}
\CommentTok{\# Running the function}
\FunctionTok{GetCellLineDAFun}\NormalTok{(CNV.CellLine,}\AttributeTok{ncores=}\DecValTok{2}\NormalTok{, CTDDirectory)}
\DocumentationTok{\#\# [1] "cluster making started"}
\DocumentationTok{\#\# NULL}
\CommentTok{\# Output}
\FunctionTok{load}\NormalTok{(}\FunctionTok{paste0}\NormalTok{(CTDDirectory,}\StringTok{"/CellLine\_DA\_genes/DA\_Gene/LK2\_LUNG\_cell\_dm\_genes.Rda"}\NormalTok{))}
\NormalTok{knitr}\SpecialCharTok{::}\FunctionTok{kable}\NormalTok{(cell\_da\_genes[}\DecValTok{1}\SpecialCharTok{:}\DecValTok{2}\NormalTok{,], }\AttributeTok{caption =} \StringTok{\textquotesingle{}DA genes of the LK2 lung cancer cell line\textquotesingle{}}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\begin{table}

\caption{\label{tab:unnamed-chunk-83}DA genes of the LK2 lung cancer cell line}
\centering
\begin{tabular}[t]{l}
\hline
V1\\
\hline
SIGLEC14\\
\hline
CDKN2B-AS1\\
\hline
\end{tabular}
\end{table}

Computing sample-specific DA genes: To compute sample-specific DA genes, users need to compute highly amplified and deleted genes utilizing chromosomal copy number profiles of each cancer type cohort as inputs in the GISTIC 2.0 tool in GenePattern web server (\url{https://www.genepattern.org/}) using a confidence interval of 0.90. They should select the genes that exceed the high-level GISTIC thresholds for amplification and deletions as 2 and -2, respectively as DA genes.

\hypertarget{step-5-computing-sample-cell-line-pathway-activity-based-similarity-score.}{%
\subsection{\texorpdfstring{ Step 5: Computing sample-cell line pathway activity-based similarity score.}{  Step 5: Computing sample-cell line pathway activity-based similarity score.}}\label{step-5-computing-sample-cell-line-pathway-activity-based-similarity-score.}}

\hypertarget{function1-findsimfun}{%
\subsubsection{\texorpdfstring{ Function1: FindSimFun}{  Function1: FindSimFun}}\label{function1-findsimfun}}

\begin{Shaded}
\begin{Highlighting}[]
\NormalTok{?FindSimFun}
\DocumentationTok{\#\# i Rendering development documentation for \textquotesingle{}FindSimFun\textquotesingle{}}
\CommentTok{\# Running the function to compute DNA methylation based Spearman similarity score of a patient{-}cell line pair}
\FunctionTok{data}\NormalTok{(}\StringTok{"CTDPathSim2"}\NormalTok{)}
\NormalTok{x}\OtherTok{\textless{}{-}}\FunctionTok{FindSimFun}\NormalTok{(pat\_dm\_genes,pat\_reactome,Deconv\_meth\_gene,cell\_dm\_genes,cell\_reactome,ccle\_methylation)}
\NormalTok{knitr}\SpecialCharTok{::}\FunctionTok{kable}\NormalTok{(x, }\AttributeTok{caption =} \StringTok{\textquotesingle{}DNA methylation based Spearman similarity score of a patient{-}cell line pair\textquotesingle{}}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\begin{table}

\caption{\label{tab:unnamed-chunk-84}DNA methylation based Spearman similarity score of a patient-cell line pair}
\centering
\begin{tabular}[t]{r}
\hline
Spearman\_Similarity\_Score\\
\hline
0.6796275\\
\hline
\end{tabular}
\end{table}

\begin{Shaded}
\begin{Highlighting}[]
\CommentTok{\# Running the function to compute gene expression{-}based Spearman similarity score of a patient{-}cell line pair}
\NormalTok{y}\OtherTok{\textless{}{-}}\FunctionTok{FindSimFun}\NormalTok{(pat\_de\_genes,pat\_reactome,Deconv\_expression,cell\_de\_genes,cell\_reactome,ccle\_expr)}
\NormalTok{knitr}\SpecialCharTok{::}\FunctionTok{kable}\NormalTok{(y, }\AttributeTok{caption =} \StringTok{\textquotesingle{}Gene expression{-}based Spearman similarity score of a patient{-}cell line pair\textquotesingle{}}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\begin{table}

\caption{\label{tab:unnamed-chunk-84}Gene expression-based Spearman similarity score of a patient-cell line pair}
\centering
\begin{tabular}[t]{r}
\hline
Spearman\_Similarity\_Score\\
\hline
0.1575758\\
\hline
\end{tabular}
\end{table}

\begin{Shaded}
\begin{Highlighting}[]
\CommentTok{\# Running the function to compute copy number value based Spearman similarity score of a patient{-}cell line pair}
\NormalTok{z}\OtherTok{\textless{}{-}}\FunctionTok{FindSimFun}\NormalTok{(pat\_da\_genes,pat\_reactome,pat\_cnv,cell\_da\_genes,cell\_reactome,ccle\_cnv)}
\NormalTok{knitr}\SpecialCharTok{::}\FunctionTok{kable}\NormalTok{(z, }\AttributeTok{caption =} \StringTok{\textquotesingle{}Copy number aberration{-}based Spearman similarity score of a patient{-}cell line pair\textquotesingle{}}\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\begin{table}

\caption{\label{tab:unnamed-chunk-84}Copy number aberration-based Spearman similarity score of a patient-cell line pair}
\centering
\begin{tabular}[t]{r}
\hline
Spearman\_Similarity\_Score\\
\hline
0.0966884\\
\hline
\end{tabular}
\end{table}

For CTDPathSim2.0, after computing the three different scores for all the sample-cell line pairs, the users need to scale the Spearman similarities to the range of 0 to 1 using min-max normalization and should compute an average similarity score taking a mean of expression-based score, DNA methylation-based score, and copy number-based score for each sample-cell line pair.

For CTDPathSim1.0, the users should perform the above step with only expression-based score and DNA methylation-based score.


\end{document}
